### PR TITLE
Allow creation of multiple code generators.

### PR DESCRIPTION
**Pull request by prajwalkkumar**

### CODE REVIEW

File: code_generator_funcadl_xAOD/pyproject.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ packages = [{include = "code_generator_funcadl_xaod"}]
 [tool.poetry.dependencies]
 python = "~3.10"
 func-adl-xAOD = "2.0.1"
-servicex-code-gen-lib = {version = "^1.1.3"}
```
Comment (BenGalewsky) [inline code_generator_funcadl_xAOD/pyproject.toml:12]: Any reason this doesn't use 1.1.4a11 like the uproot transformer?

File: helm/servicex/templates/app/configmap.yaml [hunk 2]
```diff
@@ -147,10 +146,25 @@ data:
...
+         {{- $codeGenImage := printf "%s:%s" .image .tag }}
+         {{- $dictItem := printf "\"%s\":\"%s\"" $codeGenName $codeGenImage }}
+         {{- $code_gen_images = append $code_gen_images $dictItem }}
+         {{- $image := printf "%s:%s" .defaultScienceContainerImage .defaultScienceContainerTag }}
+         {{- $dictItem := printf "\"%s\":\"%s\"" $codeGenName $image }}
+         {{ $transformer_images = append $transformer_images $dictItem }}
+    {{- end }} 
+    CODE_GEN_SERVICE_URLS = { {{ join "," $code_gen_service_urls }} }
+    CODE_GEN_IMAGES = { {{ join "," $code_gen_images }} }
+    TRANSFORMER_SCIENCE_IMAGES = { {{ join "," $transformer_images }} }
```
Comment (BenGalewsky) [inline helm/servicex/templates/app/configmap.yaml]: Does ScienceImage belong here? Shouldn't this be served up from the code gen itself?

**Comment by BenGalewsky** [comment]
I tried deploying the helm chart with these values:
```
codeGen:
  uproot:
    enabled: true
    image: sslhep/servicex_code_gen_func_adl_uproot
    pullPolicy: Always
    tag: multiple-codegen
    defaultScienceContainerImage: sslhep/servicex_func_adl_uproot_transformer
    defaultScienceContainerTag: develop
  python:
    enabled: true
    image: sslhep/servicex_code_gen_python
    pullPolicy: Always
    tag: multiple-codegen
    defaultScienceContainerImage: sslhep/servicex_func_adl_xaod_transformer
    defaultScienceContainerTag: develop
  atlasxaod:
    enabled: false
  cms:
    enabled: false
```
And I get this error from helm:
```
Error: template: servicex/templates/codegen/service.yaml:3:7: executing "servicex/templates/codegen/service.yaml" at <.enabled>: can't evaluate field enabled in type interface {}
```

File: servicex_app/servicex/code_gen_adapter.py [hunk 3]
```diff
@@ -31,45 +31,62 @@
...
         :returns a tuple of (config map name, default transformer image)
         """
         from io import BytesIO
         from zipfile import ZipFile
 
         assert self.transformer_manager, "Code Generator won't work without a Transformer Manager"
 
+        # Finding Codegen URL from the config dictionary and user provided input
+        post_url = ''
+        for key, value in self.code_gen_service_urls.items():
```
Comment (BenGalewsky) [inline servicex_app/servicex/code_gen_adapter.py]: Why isn't this just 
```
post_url = self.code_gen_service_urls.get(user_codegen_name, None)
```

File: helm/servicex/templates/app/configmap.yaml (hunk 2 above)
Comment (BenGalewsky) [reply helm/servicex/templates/app/configmap.yaml]: Reminder on this one...

File: servicex_app/servicex/resources/transformation/submit.py [hunk 4]
```diff
@@ -96,9 +97,20 @@ def post(self):
...
             config = current_app.config
 
-            image = args["image"]
+            image = args.get("image")
             did = args.get("did")
             file_list = args.get("file-list")
+            user_codegen_name = args.get("codegen")
+
+            code_gen_image_name = ''
+            for key, val in config['CODE_GEN_IMAGES'].items():
```
Comment (BenGalewsky) [inline servicex_app/servicex/resources/transformation/submit.py]: Same here:
```
code_gen_image_name = config['CODE_GEN_IMAGES'].get(user_codegen_name, None)
```

File: servicex_app/servicex/resources/transformation/submit.py [hunk 5]
```diff
@@ -96,9 +97,20 @@ def post(self):
...
+                if key == user_codegen_name:
+                    code_gen_image_name = val
+
+            print("Before Code gen Image Name: ", code_gen_image_name)
```
Comment (BenGalewsky) [inline servicex_app/servicex/resources/transformation/submit.py]: Get rid of development print statements.

**Review by BenGalewsky — CHANGES_REQUESTED** [review]
Also, please add a test for submitting a transform request with an invalid code-gen setting.

File: code_generator_funcadl_xAOD/pyproject.toml (hunk 1 above)
Comment (shriram192) [reply code_generator_funcadl_xAOD/pyproject.toml:12]: could you merge the codegen lib to develop first? we will build a 1.1.5 release there and then bump all the versions here?

Comment (shriram192) [reply code_generator_funcadl_xAOD/pyproject.toml:12]: I can see only 1.1.4-alpha-14 in all versions in the latest commit

File: servicex_app/servicex/code_gen_adapter.py (hunk 3 above)
Comment (shriram192) [reply servicex_app/servicex/code_gen_adapter.py]: This has been changed

File: servicex_app/servicex/resources/transformation/submit.py (hunk 4 above)
Comment (shriram192) [reply servicex_app/servicex/resources/transformation/submit.py]: This has been changed

File: servicex_app/servicex/resources/transformation/submit.py (hunk 5 above)
Comment (shriram192) [reply servicex_app/servicex/resources/transformation/submit.py]: Removed Print statements

**Review by BenGalewsky — APPROVED** [review]
This is it! Going to cut a 1.1.4-alpha 1 release and test before merging

**Pull request merged**